yt/frontends/halo_catalog/data_structures.py

```diff
@@ -93,10 +90,17 @@ def _read_particle_positions(self, ptype, f=None):
         return pos
 
 
-class HaloCatalogDataset(SavedDataset):
+class YTHaloCatalogDataset(SavedDataset):
```

Comment:
Not from this PR but this inheritance confuses me. What does `YTHaloCatalogDataset` have in common with `SavedDataset`, conceptually ?